contracts/allocators/Kp3rAllocatorV2.sol

```diff
@@ -0,0 +1,228 @@
+    function create_lock(uint256 value_, uint256 unlockTime_) external;
+
+    function increase_amount(uint256 value_) external;
+
+    function increase_unlock_time(uint256 unlockTime_) external;
+
+    function locked(address addr_) external view returns (uint128, uint256);
+
+    function locked__end(address addr_) external view returns (uint256);
+
+    function withdraw() external;
+}
+
+interface IGauge {
+    function vote(address[] calldata tokenVote_, uint256[] calldata weights_) external;
+}
+
+interface IClaim {
+    function approve(address spender_, uint256 amount_) external returns (bool);
+
+    function claim() external returns (uint256);
+
+    function redeem(uint256 id_) external;
+
+    struct option {
+        uint256 amount;
+        uint256 strike;
+        uint256 expirty;
+        bool exercised;
+    }
+
+    function options(uint256 id_) external view returns (option memory);
+}
+
+// errors
+error Kp3rAllocatorV2_InvalidAddress();
+error Kp3rAllocatorV2_LowUSDCLimits();
+
+contract Kp3rAllocatorV2 is BaseAllocator {
+    uint256 private constant MAX_TIME = 4 * 365 * 86400 + 1;
+
+    ITreasury public treasury;
+
+    IKp3rVault public kp3rVault;
+    IGauge public gauge;
+    IClaim public distributor;
+    IClaim public rKp3r;
+
+    uint256 lockEnd;
+    uint128 usdcUsed;
+
+    constructor(
+        AllocatorInitData memory data,
+        address treasury_,
+        address kp3rVault_,
+        address gauge_,
+        address distributor_,
+        address rKp3r_
+    ) BaseAllocator(data) {
+        if (
+            treasury_ == address(0) ||
+            kp3rVault_ == address(0) ||
+            gauge_ == address(0) ||
+            distributor_ == address(0) ||
+            rKp3r_ == address(0)
+        ) revert Kp3rAllocatorV2_InvalidAddress();
+
+        treasury = ITreasury(treasury_);
+
+        kp3rVault = IKp3rVault(kp3rVault_);
+        gauge = IGauge(gauge_);
+        distributor = IClaim(distributor_);
+        rKp3r = IClaim(rKp3r_);
+
+        // approve veKP3R
+        _tokens[0].approve(address(kp3rVault), type(uint256).max);
+        _tokens[1].approve(address(rKp3r), type(uint256).max);
+
+        // approve extender
+        _tokens[0].approve(address(treasury), type(uint256).max);
+        _tokens[1].approve(address(treasury), type(uint256).max);
+        rKp3r.approve(address(treasury), type(uint256).max);
+    }
+
+    function _activate() internal override {
+        if (extender.getAllocatorLimits(2).loss < 50000000000) revert Kp3rAllocatorV2_LowUSDCLimits();
+    }
+
+    function _update(uint256 id) internal override returns (uint128 gain, uint128 loss) {
+        if (id == 2) {
+            loss = usdcUsed;
+        } else {
+            uint256 balance = _tokens[0].balanceOf(address(this));
+            (uint256 lockedBalance, ) = kp3rVault.locked(address(this));
+
+            if (balance > 0 && lockedBalance == 0) {
+                lockEnd = block.timestamp + MAX_TIME;
```

Comment:
So, a setter for time. I think that would be ok. It could also just scale below... so when it extends time that it also extends just by the same amount, since after the first lock it will anyways turn into the extend lock value.

Reply:
added an onlyGuardian setter for MAX_TIME